codex/merkletree/merkletree.nim

```diff
@@ -0,0 +1,128 @@
+import std/sequtils
+import std/math
+import std/bitops
+import std/sugar
+
+import pkg/libp2p
+import pkg/stew/byteutils
+
+import pkg/questionable
+import pkg/questionable/results
+
+
+type
+  MerkleHash* = MultiHash
+  MerkleTree* = ref object of RootObj
+    nodes: seq[seq[MerkleHash]]
+  MerkleProof* = ref object of RootObj
+    index*: int
+    path*: seq[MerkleHash]
```

Comment:
Might it be wise to make the MerkleProof index and path read-only? You know: A constructor to set them, and then read-only procs to read them.

Reply:
Makes sense, adding accessor methods 👍